### DISCUSSION TITLE
How can I send photos group as an album in telegram bot?

### DISCUSSION THREAD

**Discussion by arpitgoyall** [python-telegram-bot/python-telegram-bot#2614] [Q&A]
I want to send an album if it takes more than one URL in parameter, I saw send_media_group function but i am little bit confused that how to use it please anyone can tell me. I am dropping some lines of code for reference

bot.send_media_group(
                chat_id=update.effective_chat.id,
                media=[InputMediaPhoto(Url1, Url2)]
            )
But in url2 it reads as a caption parameter, how can I solve that problem, please help! [Function link](https://python-telegram-bot.readthedocs.io/en/stable/telegram.bot.html?highlight=send%20album#telegram.Bot.send_media_group)

**Comment by Poolitzer** [comment]
Hey man, you were very close :D The correct call would be

`bot.send_media_group(
chat_id=update.effective_chat.id,
media=[InputMediaPhoto(Url1), InputMediaPhoto(Url2)]
)`

**Reply by arpitgoyall** [reply]
Thanks, bro it worked now the problem I am getting is whenever I set caption it goes in each media separately but I want to keep it universal for all images how can I do that because the separate caption is only showing by clicking on images.

**Reply by harshil21** [reply]
I think if you set the caption for the first image, then it'll show for all.

**Reply by Poolitzer** [reply]
That is correct

**Reply by arpitgoyall** [reply]
Yes, that is working perfectly but can anyone tell me how to do it with a code.
            
            ImageUrls = [] #list that can contain unknown number of urls
            list = ImageUrls
            bot.send_media_group(
            chat_id=update.effective_chat.id,
            media=[InputMediaPhoto(list[0], caption="Anything"), InputMediaPhoto(list[i] for i in range(1, len(list)))]
            )
raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type generator is not JSON serializable

I tried that code but it is giving an error, does anyone have a solution for it to set caption to first image only?

**Reply by Poolitzer** [reply]
You dont generate several objects of InputMediaPhoto, you generate one where the first argument is said type generator. You can change that to `media=[InputMediaPhoto(list[0], caption="Anything")] +  [InputMediaPhoto(list[i] for i in range(1, len(list))]` to have it in one line and working

**Reply by arpitgoyall** [reply]
Thank you so much it worked but that syntax would not work we have to use for loop outside the input like:

`media=[InputMediaPhoto(list[0],caption=update.effective_message.text)] +[InputMediaPhoto(list[i]) for i in range(1, len(list))]`

**Reply by Poolitzer** [reply]
uh right I forgot the closing braket. Good fix!
